README.md

```diff
@@ -10,16 +23,78 @@ The agent is used through the [StudyBits Wallet](https://github.com/Quintor/Stud
 
 Contact us on [Gitter](https://gitter.im/StudyBits/Lobby)
 
-## Running in docker
 
-Use `TEST_POOL_IP=127.0.0.1 docker-compose up --build --force-recreate pool university-agent-rug university-agent-gent` 
+## Contributing <a name="contributing"></a>
+
+### Repository Structure <a name="structure"></a>
+
+The StudyBits repo consists of the following parts:
+
+```
++ ci
+    + Contains docker files used by trevor for ci.
++ docs
+    + Contains design and implementation documentation.
++ university-agent
+    + Contains java source-code for the university agent.
+```
+
+### Development Setup <a name="dev"></a>
+
+In order to develop for StudyBits a couple of dependencies need to be installed:
+
+1. Install [libindy](https://github.com/hyperledger/indy-sdk/tree/master/libindy)
+    The current version of StudyButs requires Libindy v1.6.6 to be installed, note that later versions are not supperted currently.
+
+2. Make sure [Project Lombok](https://projectlombok.org/) can work it's magic. In your IDE of choice, please enable precompiling.
+
+### How to send a PR <a name="pullrequests"></a>
+
+Before sending a PR make sure to be complient with the following:
 
-## Running tests in docker
++ Do not create many PRs for one feature. Consider implementing a complete feature before sending a PR. 
++ Consider sending a design doc folder for a new feature.
```

Comment:
Maybe replace this with "if design is needed, create an issue to discuss before starting to code"?